sdk/python/feast/infra/provider.py

```diff
@@ -0,0 +1,44 @@
+import abc
+from typing import List
+
+from feast import FeatureTable
+from feast.repo_config import FirestoreConfig, OnlineStoreConfig
+
+
+class Provider(abc.ABC):
```

Comment:
Can you define a `provider` please? It seems like it is synonymous with an online store here. The original idea was that it is either related to the whole environment (`aws`) or related to your compute, not a specific store.